src/main/java/control/ControllerManager.java

```diff
@@ -146,7 +148,7 @@ private void handleCancel(MouseEvent event) {
      *
      * @param block ShotBlock to set as active
      */
-    public void setActiveShotBlock(ShotBlock block) {
+    public void setActiveShotBlock(Optional<ShotBlock> block) {
```

Comment:
The `null` check could very well be done inside the setter, allowing you to set it to `null`, resulting in an empty optional.